sphinx/texinputs/sphinx.sty

```diff
@@ -76,40 +76,62 @@
       #1\dimexpr\linewidth-\arrayrulewidth\relax-\tw@\tabcolsep-\arrayrulewidth\relax}}
 % using here T (for Tabulary) feels less of a problem than the X could be
 \newcolumntype{T}{J}%
+% For tables allowing pagebreaks
 \RequirePackage{longtable}
-% For table captions.
-\RequirePackage{threeparttable}
-% fixing the LaTeX mess of vertical spaces with threeparttable and longtable
-% The user interface:
+% User interface to set-up whitespace before and after tables:
 \newcommand*\sphinxtablepre {0pt}%
 \newcommand*\sphinxtablepost{\medskipamount}%
+\newcommand*\sphinxbelowcaptionspace{.5\sphinxbaselineskip}%
 % as one can not use \baselineskip from inside longtable (it is zero there)
 % we need \sphinxbaselineskip, which defaults to \baselineskip
-\newcommand*\sphinxbelowcaptionspace{.5\sphinxbaselineskip}%
 \def\sphinxbaselineskip{\baselineskip}%
-% Helper macros, not a priori for user customization
+% These commands are inserted by the table templates
 \def\sphinxatlongtablestart
    {\par
     \vskip\parskip
     \vskip\dimexpr\sphinxtablepre\relax % adjust vertical position
     \vbox{}% get correct baseline from above
     \LTpre\z@skip\LTpost\z@skip % set to zero longtable's own skips
-    \edef\sphinxbaselineskip{\dimexpr\the\dimexpr\baselineskip\relax\relax}}%
+    \edef\sphinxbaselineskip{\dimexpr\the\dimexpr\baselineskip\relax\relax}%
+   }%
 \def\sphinxatlongtableend{\prevdepth\z@\vskip\sphinxtablepost\relax}%
-% the longtable template inserts a \strut at caption's end
 \def\sphinxlongtablecapskipadjust
-   {\dimexpr-\dp\strutbox-\sphinxbaselineskip
-            +\sphinxbelowcaptionspace\relax}%
-% tabular(y) with or without threeparttable
+   {\dimexpr-\dp\strutbox-\sphinxbaselineskip+\sphinxbelowcaptionspace\relax}%
+% Now for tables not using longtable
 \def\sphinxattablestart
    {\par
     \vskip\dimexpr\sphinxtablepre\relax
-    \belowcaptionskip\sphinx@TPTbelowcaptionskip}%
+   }%
 \let\sphinxattableend\sphinxatlongtableend
-% the tabular(y) templates use [t] vertical placement parameter
-\def\sphinx@TPTbelowcaptionskip
-   {\dimexpr-1.2\baselineskip % .2\baselineskip hardcoded in threeparttable
-            +\sphinxbelowcaptionspace\relax }%
+% longtable's wraps captions to a maximal width of \LTcapwidth
+% so we do the same for all tables
+\newcommand\sphinxcaptionof[3][\LTcapwidth]{%
+   \noindent\hb@xt@\linewidth{\hss
+      \vtop{\@tempdima\dimexpr#1\relax
+% don't exceed linewidth for the caption width
+            \ifdim\@tempdima>\linewidth\hsize\linewidth\else\hsize\@tempdima\fi
+            \def\@captype{#2}%
+            \capstart % must be after setting \@captype
```

Comment:
Actually this is good if caption is above table. But if user positions in table template the `\sphinxcaptionof` below table, this means a link will bring here, not at start of table. Thus I should abstract and separate a `\sphinxtablecapstart` for more flexibility.